src/binary/writer.rs

```diff
@@ -542,58 +675,25 @@ impl<W: Write> BinarySystemWriter<W> {
         })
     }
 
-    pub fn write_clob(&mut self, value: &[u8]) -> IonResult<()> {
+    fn write_clob<A: AsRef<[u8]>>(&mut self, value: A) -> IonResult<()> {
```

Comment:
The trait makes `write_clob` and `write_blob` generic over anything that can be viewed as a byte sequence. This allows you to call this method with a `&str`, `String`, `Vec<u8>`, etc.